tools/print_test_stats.py

```diff
@@ -108,6 +108,13 @@ def plural(n: int) -> str:
     return '' if n == 1 else 's'
 
 
+def get_base_commit(sha1: str) -> str:
+    return subprocess.check_output(
+        ["git", "merge-base", sha1, "origin/master"],
+        encoding="ascii",
+    ).strip()
```

Comment:
I don't think this is correct, is it? Specifically, I believe this is only correct if the given `sha1` is not on `master`

Reply:
Ah does that mean we'd need to check the current branch first? If it's on master, it would just be the parent commit right?

Reply:
Wait I think this does work? If both the commits were on master, wouldn't it give the earlier commit?

Reply:
This is simply a repositioned code in print_regressions. also this field is useless unless build_pr is non-null anyway.



if `sha1` came from master already this will return the same result as the input `sha1`.
I think nightly would have problems since those are bundled commits from previous nightly and could cause problem.

Reply:
I guess i dont have a good "correct" definition if the sha1 doesn't come from a PR or a branch. if there's any better definition I would love to incorporate.

Reply:
If you're only planning to use the `build_base_commit` field when it's on a PR then I guess this is fine. However, the logic to correct for the case where it is on `master` is pretty straightforward:

https://github.com/pytorch/pytorch/blob/935057fc7464d0df6741ffc24d5aed3131533073/tools/print_test_stats.py#L808-L813

Can you just use that same logic here?

Reply:
oh I see. yeah in my case I actually want the `base` to be the same as the `sha1` for scribe report. since there's no branch off so the branch off point is itself. different from the comparison step, where it is trying to find the previous commit to compare CI with.

Reply:
Ok, if that's what you want then yeah it's fine as-is :)